### PR TITLE
Loose constraint for `path-io`

### PR DESCRIPTION
**Pull request by juanpaucar**
This fixes: https://github.com/stackbuilders/hapistrano/issues/73

### CODE REVIEW

**Comment by mrkkrp** [comment]
You should not change `stripDir` to `stripProperPrefix` like this. You need to use CPP to choose between the two depending on version of `path`. Otherwise you're making it work only with `path-0.6.0` and later and lose compatibility with older versions.

**Comment by mrkkrp** [comment]
See for example here: https://github.com/mrkkrp/path-io/blob/master/Path/IO.hs#L396-L400.

**Comment by juanpaucar** [comment]
Thanks for the headsup @mrkkrp. I didn't realize [constraints for path](https://github.com/stackbuilders/hapistrano/blob/master/hapistrano.cabal#L49) are oddly written. I'll also change that

**Comment by mrkkrp** [comment]
Uh, yeah, should be `<= 0.7` now I guess, certainly not `<= 6.0`!

File: src/System/Hapistrano.hs [hunk 1]
```diff
@@ -201,7 +200,11 @@ deployedReleases
 deployedReleases deployPath = do
   let rpath = releasesPath deployPath
   xs <- exec (Find 1 rpath :: Find Dir)
+#if MIN_VERSION_path(0,6,0)
```
Comment (sestrella) [inline src/System/Hapistrano.hs]: @juanpaucar I think you need to extract this out to one place:

```HS
stripDirs :: FilePath -> [FilePath] -> m [FilePath]
stripDirs path dirs =
#if MIN_VERSION_path(0,6,0)
  mapM (stripProperPrefix path) dirs
#else
  mapM (stripDir path) dirs
#endif
```

File: src/System/Hapistrano.hs [hunk 2]
```diff
@@ -213,7 +216,11 @@ completedReleases
 completedReleases deployPath = do
   let cpath = ctokensPath deployPath
   xs <- exec (Find 1 cpath :: Find File)
+#if MIN_VERSION_path(0,6,0)
```
Comment (sestrella) [inline src/System/Hapistrano.hs]: @juanpaucar you should be able to reuse `stripDirs` here

**Pull request merged**